lib/std/Uri.zig

```diff
@@ -0,0 +1,515 @@
+//! Implements URI parsing roughly adhering to <https://tools.ietf.org/html/rfc3986>.
+//! Does not do perfect grammar and character class checking, but should be robust against URIs in the wild.
+
+const Uri = @This();
+const std = @import("std.zig");
+const testing = std.testing;
+
+scheme: ?[]const u8,
```

Comment:
scheme is not optional, all forms of URI require a scheme (just like they require a path)

Reply:
what about host?

Reply:
URI encompass URNs, which have no authority. The most simple URI is `scheme:something` where `something` is the "path"